### ISSUE TITLE
Confusion regarding fmt.rs `{:.*}` syntax

### ISSUE THREAD

**Issue by leonardinius** [rust-lang/rust#22927]
See IRC discussion starting from https://botbot.me/mozilla/rust/2015-03-01/?msg=33094439&page=16

---
tl;dr;

- See http://doc.rust-lang.org/std/fmt/index.html#argument-types
- It introduces `{:.*}` syntax without proper explanation what it is.
- It's confusing, some think it's regexp, other think (e.g. me) it's not
- It does not specify behavior and doesn't provide minimal samples. 

It might be that some kind of explanation similar to https://botbot.me/mozilla/rust/msg/33094764/ is required; as well as at least few examples, such as https://botbot.me/mozilla/rust/msg/33094776/, as well as this https://botbot.me/mozilla/rust/msg/33094947/

I 'm afraid I didn't completely grok examples featuring `$` from the IRC conversation, but I think they should be explained as well.

**Comment by theemathas** [comment]
Another documentation problem in fmt: How is width defined for unicode strings?

Current behavior: `println!("a{:5}b", "n͈̰̎")` outputs `an͈̰̎ b`

**Comment by theemathas** [comment]
Related to https://github.com/rust-lang/rust/issues/22646
